src/sql/parts/jobManagement/views/jobHistory.component.ts

```diff
@@ -144,7 +151,23 @@ export class JobHistoryComponent extends Disposable implements OnInit {
 
 	private jobAction(action: string, jobName: string): void {
 		let ownerUri: string = this._dashboardService.connectionManagementService.connectionInfo.ownerUri;
-		this._jobManagementService.jobAction(ownerUri, jobName, action);
+		const self = this;
+		this._jobManagementService.jobAction(ownerUri, jobName, action).then(result => {
+			if (result.succeeded) {
+				switch (action) {
+					case ('run'):
```

Comment:
not sure how permanent this code is, but it would be better to use an enum here.